core/domain/base_model_validators.py

```diff
@@ -110,6 +110,13 @@ class BaseModelValidator(python_utils.OBJECT):
     # model for which the external model is being fetched. Each value consists
     # of a list of ExternalModelReference objects.
     field_name_to_external_model_references = collections.defaultdict(list)
+    # Domain validation for some models provides an option of strict vs non
+    # strict check. Currently the models which provide this feature are
+    # collection model and exploration model.
+    # In cases when models are not published, domain validation should be
+    # non strict. The subclasses should override this var whenever the
+    # validation should be non strict.
+    use_non_strict_domain_validation = False
```

Comment:
Seems confusing to use a global, especially if this is supposed to vary on a per-instance basis. Why can't the job pass it through to the relevant function instead?

Reply:
The way the domain object validation is performed is via [_validate_model_domain_object_instances](https://github.com/oppia/oppia/blob/develop/core/domain/base_model_validators.py#L171) being called in the validate method of the base class: https://github.com/oppia/oppia/blob/develop/core/domain/base_model_validators.py#L333.

Using a new argument in the `_validate_model_domain_object_instances` method would create the issue that we will need to override the validate method for exploration and collection models which would break the regular structure. So, I used a separate boolean to configure the same.

Reply:
I think it is reasonable to pass in **kwargs flags that should then be passed to validate(), or have a global convention for the structure of validate() in each model (e.g. it should only accept at most one param called "strict", and by default it is True). What do you think? Maybe discuss with @vojtechjelinek?

Reply:
Hmm, the plan sounds good but that will be a large refactor and should be handled separately. I want this PR to fix the existing issues in jobs - should I file a separate issue for the refactor?

Reply:
I don't think we should introduce this pattern where you're using a global -- again, maintainability is a concern here. 

Is there an alternative way to fix the jobs without using a global? If not, we should still find a way to do this one properly (which may involve some refactoring), but the other fixes in this PR can still go in.

Reply:
Hmm, one of the ways to handle it in this PR would be to override the validate method for collection and exploration models. Let me look into the refactor and if it seems doable in this one itself, I can take it up otherwise will look for other approach.

Reply:
Hmm, also one issue is that the validate method is called via the prod jobs and not by the validator, so it would be not clean to check in the job if a model is private or public.

Reply:
What do you think about overriding the method `_validate_model_domain_object_instances` method?

Reply:
If we want that to behave in a standard way for all model classes (i.e. call validate()), I think we should generalize the existing functionality rather than overwrite it. What do you think?

Reply:
Updated as per offline discussion.